[DOCU-2617] Cherry-pick of secrets management in konnect (#4828) (#4845)
Kong/docs.konghq.com: app/konnect/runtime-manager/vaults/index.md

-------|------|------------
1 | **Vaults menu link** | Main link to the vaults configuration for a runtime group. Appears when you select a runtime group.
2 | **Add vault** | Click the **Add vault** button to set up any supported Konnect vault backend.
3 | **Vault entry** | Select a vault entry to open the configuration page for the particular vault. On each vault's configuration page, you can edit or delete the vault, or copy the entire configuration as JSON.
4 | **Vault ID** | The vault's UUID. 
5 | **Vault action menu** | From this menu, you can <b>view</b>, <b>edit</b>, or <b>delete</b> a vault's configuration. 

## Use cases

Vaults have several use cases: 
* Storing secrets securely
* Managing access to secrets with fine-grained policies
* Applying internal security policies
* Automating secret rotation
* Auditing secrets usage
* Encryption of secrets at rest

{{site.konnect_short_name}} **does not**:
* Store credentials to access the vault itself.
You must provide those credentials to the {{site.base_gateway}} data plane directly.
* Update or modify the secrets in 3rd party vaults.

Vaults are configurable per runtime group. You can't use the same vault across

> multiple runtime groups.

Reviewer: Can you clarify what this means? Are we trying to say that the vault config is runtime group specific and  if the user wants the same vault config in multiple groups, the vault config needs to be added in every runtime group?

I am interpretting this line to mean that I cannot apply AWS Secrets Manager to more than one runtime group, which is not accurate.

Reviewer 2: @smritikjaggi I think that's what Lena was trying to say here, that the vault config is runtime group-specific. 

Would something like this be clearer that you can apply AWS secrets manager to more than one runtime group?

> Vault configurations are runtime group-specific. If you want to use the same vault configuration across
multiple runtime groups, you must configure it for each group individually.

> Vaults are configurable per runtime group. You can't use the same vault across
> multiple runtime groups.

Reviewer 3: ```suggestion
Vault configurations are runtime group-specific. If you want to use the same vault configuration across
multiple runtime groups, you must configure it for each group individually.
```

Reviewer 3: @cloudjumpercat , I'd reviewed your suggestions w/ @smritikjaggi on a call and agree that's clearer. I've added the changes. Thank you!.

## Supported vaults in {{site.konnect_short_name}}

Konnect supports the following vault backends:
* AWS Secrets Manager
* HashiCorp Vault
* GCP Secret Manager
* Environment variables

You can manage all of these vaults through the [Runtime Manager](/konnect/runtime-manager/vaults/how-to) or with [decK](/deck/latest/guides/vaults/).

## See also

Check out the example use case for [storing certificates in a vault](/konnect/runtime-manager/vaults/how-to).

For detailed vault configuration references and guides, see the {{site.base_gateway}}
documentation:

> * [AWS Secrets Manager](/gateway/{{page.kong_version}}/kong-enterprise/secrets-management/backends/aws-sm)

Reviewer: Looks like the link is broken for all of these detailed guides

Reviewer 3: s/b https://docs.konghq.com/gateway/3.0.x/kong-enterprise/secrets-management/backends/aws-sm/

Reviewer 2: Yeah, I think this is having issues with rendering a Gateway docs page because Konnect pages aren't versioned, so it doesn't have a page version to pull from. I think this link should be `/latest`.

Reviewer 3: Thanks!

> * [AWS Secrets Manager](/gateway/{{page.kong_version}}/kong-enterprise/secrets-management/backends/aws-sm)

Reviewer 2: ```suggestion
* [AWS Secrets Manager](/gateway/latest/kong-enterprise/secrets-management/backends/aws-sm)
```

> * [GCP Secrets Manager](/gateway/{{page.kong_version}}/kong-enterprise/secrets-management/backends/gcp-sm)

Reviewer 2: ```suggestion
* [GCP Secrets Manager](/gateway/latest/kong-enterprise/secrets-management/backends/gcp-sm)
```

> * [HashiCorp Vault](/gateway/{{page.kong_version}}/kong-enterprise/secrets-management/backends/hashicorp-vault)

Reviewer 2: ```suggestion
* [HashiCorp Vault](/gateway/latest/kong-enterprise/secrets-management/backends/hashicorp-vault)
```

> * [Environment variables](/gateway/{{page.kong_version}}/kong-enterprise/secrets-management/backends/env)

Reviewer 2: ```suggestion
* [Environment variables](/gateway/latest/kong-enterprise/secrets-management/backends/env)
```
